staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -146,10 +146,13 @@ func (cnc *CloudNodeController) Run(stopCh <-chan struct{}) {
 
 // UpdateNodeStatus updates the node status, such as node addresses
 func (cnc *CloudNodeController) UpdateNodeStatus(ctx context.Context) {
-	instances, ok := cnc.cloud.Instances()
+	var instances cloudprovider.Instances
+	instancesV2, ok := cnc.cloud.InstancesV2()
 	if !ok {
-		utilruntime.HandleError(fmt.Errorf("failed to get instances from cloud provider"))
-		return
+		if instances, ok = cnc.cloud.Instances(); !ok {
```

Comment:
Don't love the nested ifs here, can we do something like this instead:
```go
instancesV2, instanceV2Supported := cnc.cloud.InstancesV2()
instances, instancesSupported := cnc.cloud.Instances()

if instanceV2Supported == nil && instancesSupported == nil {
    utilruntime.HandleError("either Instances or InstanceV2 must be supported")
    return
}
```

Reply:
Sure, looks good.